### ISSUE TITLE
Change package id

### ISSUE THREAD

**Issue by ololoe** [Telegram-FOSS-Team/Telegram-FOSS#709]
Please change package id. It's impossible to have telegram and telegram foss without workarounds.

**Comment by Gabr-F** [comment]
I agree, I ran into the problem myself and I'd wish to have both apps installed

**Comment by Aryan-Chauhan** [comment]
True.

Telegram is showing up under Play Store app updates section, even though I'm using Telegram FOSS, not the Play Store version.

If I try to update it from Play Store, it fails, so that's good.

Having a different package name from Play Store version will solve this little problem.

**Comment by ololoe** [comment]
> Telegram is showing up under Play Store app updates section, even though I'm using Telegram FOSS, & not the Play Store version.
> 
> If I try to update it from Play Store, it fails, so that's good.

That is some android/play bug. Play doesn't see that the app is not installed by it.

**Comment by scrax** [comment]
> > Telegram is showing up under Play Store app updates section, even though I'm using Telegram FOSS, & not the Play Store version.
> > If I try to update it from Play Store, it fails, so that's good.
> 
> That is some android/play bug. Play doesn't see that the app is not installed by it.

that's surely not a bug in playstore, app ID should be unique for each app. sharing it between different version is not how android works. But seems dev of android FOSS will not fix this bug, he was already asked to do it on TG and refused

**Comment by kuesji** [comment]
it's clearly a bug in play store. app stores shouldn't assume an app came from them unless both `package name` _and_ `application signature` matches the one in their store. it's play store's fault to assume telegram foss installed by play store while telegram foss clearly signed by different store which is f-droid in this case.


also telegram and telegram foss shares same `name` and `package name` because they are same application with minor differences like getting rid of google and other proprietary dependencies. otherwise telegram foss would not exist due to trademark issues and revoked api keys.
( check transparency and breach of terms at https://core.telegram.org/api/terms )

please reach to telegram and request the changes on api usage terms if you insist on `package name` change ( and high likely `name` change as side effect ).
however, correct thing to do is reporting bug to play store.

**Comment by Gabr-F** [comment]
Feb 9, 2024 17:13:34 kuesji koesnu ***@***.***>:
>
> also telegram and telegram foss shares same *name* and *package name* because they are same application with minor differences like getting rid of google and other proprietary dependencies. otherwise telegram foss would not exist due to trademark issues and revoked api keys.
> ( check transparency and breach of terms at https://core.telegram.org/api/terms )
>
> please reach to telegram and request the changes on api usage terms if you insist on *package name* change ( and high likely *name* change as side effect ).



A lot of people would like to have both Telegram FOSS and the official Telegram installed at least at times, and would even if FOSS had only a single line of difference.

That's because FOSS is updated with a delay, many people would like to give a try to the new features, and are ok at using the Googled Telegram for a bit, to do so.

Since Telegram is cloud-based you can still do it, without a huge inconvenience, by uninstalling one app and reinstalling the other; but actually that prevents you from using secret chats with either app, since they are only local and get lost when you uninstall a client.

There's plenty of clients who do use a different application id *and* name, and the name difference is not a huge deal.

Are you completely sure that the minor differences make you compliant with the current api terms?

If it's so it might indeed make sense to keep a fork with the same appid, to have a degoogled version ready that doesn't risk api keys revocation (if it really doesn't). 

However, you should in all likelihood **discourage** it's actual use until Telegram were to block the other clients, given the problems that the identical appid causes (which are probably surprising for most users).

You could maintain a version derived from Telegram FOSS as the actual recommended version (with only the app id and name changed). 
Or you could simply point to other clients as the recommended ones (for examples others derived from FOSS), discouraging Telegram FOSS until it were the only degoogled, F-Droid published, working alternative client.

By the way, if you were actually explicitly granted a permission to use the Telegram name (from Telegram), they would almost for sure allow you to keep it if you changed (only) the app id.

I and probably many others would seriously consider using Telegram FOSS if it had its own app id, but won't until it does (after already having been burned).

---

As to the Play Store problems, it's a change on their part that also affects F-Droid apps with releases on the Play Store (https://forum.f-droid.org/t/google-play-store-tries-to-update-my-f-droid-apps/24621), so it really should be addressed to Google (although they probably did it deliberately).

**Comment by Gabr-F** [comment]
> telegram and telegram foss shares same `name` and `package name` because they are same application with minor differences like getting rid of google and other proprietary dependencies. otherwise telegram foss would not exist due to trademark issues and revoked api keys. ( check transparency and breach of terms at https://core.telegram.org/api/terms )
> 
> please reach to telegram and request the changes on api usage terms if you insist on `package name` change ( and high likely `name` change as side effect ).


A lot of people would like to have both Telegram FOSS and the official Telegram installed at least at times, and would even if FOSS had only a single line of difference.

That's because FOSS is updated with a delay, many people would like to give a try to the new features, and are ok at using the Googled Telegram for a bit, to do so.

Since Telegram is cloud-based you can still do it, without a huge inconvenience, by uninstalling one app and reinstalling the other; but actually that prevents you from using secret chats with either app, since they are only local and get lost when you uninstall a client.

There's plenty of clients which do use a different application id *and* name, and the name difference is not a huge deal.

Are you completely sure that the minor differences make you compliant with the current api terms?

If it's so it might indeed make sense to keep a fork with the same appid, to have a degoogled version ready that doesn't risk api keys revocation (if it really doesn't). 

However, you should in all likelihood **discourage** it's actual use until Telegram were to block the other clients, given the problems that the identical appid causes (which are probably surprising for most users).

You could maintain a version derived from Telegram FOSS as the actual recommended version (with only the app id and name changed). 
Or you could simply point to other clients as the recommended ones (for examples others derived from FOSS), discouraging Telegram FOSS until it were the only degoogled, F-Droid published, working alternative client.

By the way, if you were actually explicitly granted a permission to use the Telegram name (from Telegram), they would almost for sure allow you to keep it if you changed (only) the app id.

I and probably many others would seriously consider using Telegram FOSS if it had its own app id, but won't until it does (after already having been burned).

---

As to the Play Store problems, it's a change on their part that also affects F-Droid apps with releases on the Play Store (https://forum.f-droid.org/t/google-play-store-tries-to-update-my-f-droid-apps/24621), so it really should be addressed to Google (although they probably did it deliberately).

**Comment by scrax** [comment]
> telegram and telegram foss shares same `name` and `package name` because they are same application with minor differences like getting rid of google and other proprietary dependencies. otherwise telegram foss would not exist due to trademark issues and revoked api keys. ( check transparency and breach of terms at https://core.telegram.org/api/terms )
> 
> please reach to telegram and request the changes on api usage terms if you insist on `package name` change ( and high likely `name` change as side effect ).

They have just same appID, not same name or same code, nor same developer. Even just a line of code different makes two app different. That's why other app change appID for debug version. It's how android expect the appID to be used, is not intended to don't let an app install another one.

Play store just enforces how appID should be used, is not a bug is what android specification require.

For example AppVerifier has same problems to verify app that share same ID.

**Comment by Gabr-F** [comment]
> They have just same appID, not same name or same code, nor same developer. Even just a line of code different makes two app different. That's why other app change appID for debug version. It's how android expect the appID to be used, is not intended to don't let an app install another one.
> 
> Play store just enforces how appID should be used, is not a bug is what android specification require.


Almost completely agree, except that the Play Store reportedly did use to hide updates with different signatures.

And that I think it's ok to use the same appid for the Play and e.g. F-Droid releases **from the same developer** (although I prefer different ones, and for sure different signatures).

If using the same appid would prevent api keys revocation though (I haven't checked), as I said it might be ok to leave FOSS as it is, but only **as a last resort** and strongly discouraged until it's necessary.

**Comment by Gabr-F** [comment]
Ok, to be honest, what makes Telegram and FOSS different from other F-Droid/Play apps is (most of all) that the Telegram upstream source itself is released with a delay, and there are frequent interesting updates in the official beta versions.

**Comment by ololoe** [comment]
I use FOSS because it has less censorship. Some channels that are "violating laws" are not shown in original but are shown in FOSS version.

**Comment by ololoe** [comment]
Please do it quick! Registration is only possible from original Telegram! I had to delete FOSS to do it!

**Comment by Aryan-Chauhan** [comment]
Currently, I'm using 2 accounts on Telegram FOSS with a changed package name (org.telegram.messenger.foss)

Working fine as of now.

BTW, I changed the package name using ReVanced Manager.

**Comment by ololoe** [comment]
@Aryan-Chauhan 
What is with updates?

**Comment by Aryan-Chauhan** [comment]
> @Aryan-Chauhan 
> What is with updates?

I don't understand your question.

**Comment by ololoe** [comment]
> > @Aryan-Chauhan
> > What is with updates?
> 
> I don't understand your question.

How do you install updates?

**Comment by ololoe** [comment]
FUCK!!!!!!

When I deleted Telegram FOSS to register, all the links to channels were fucked up! It's all because of you!!!!

**Comment by Codwizer** [comment]
@ololoe lox

**Comment by Aryan-Chauhan** [comment]
> > > @Aryan-Chauhan
> > > What is with updates?
> > 
> > I don't understand your question.
> 
> How do you install updates?

First I download the new APK, then change it's package name, then install it on top of old app.

**Comment by ololoe** [comment]
@Codwizer xryukni, svinka!



> > > > @Aryan-Chauhan
> > > > What is with updates?
> > > 
> > > 
> > > I don't understand your question.
> > 
> > 
> > How do you install updates?
> 
> First I download the new APK, then change it's package name, then install it on top of old app.

And how do you know there is a new version? You know, you can update with F-Droid.

**Comment by ololoe** [comment]
> **FUCK!!!!!!**
> 
> **When I deleted Telegram FOSS to register, all the links to channels were fucked up! It's all because of you!!!!**

**FUCK!!!!!!! BITCH!!!! SHIT!!!!**

**Comment by Aryan-Chauhan** [comment]
> @Codwizer xryukni, svinka!
> 
> 
> 
> > > > > @Aryan-Chauhan
> > > > > What is with updates?
> > > > 
> > > > 
> > > > I don't understand your question.
> > > 
> > > 
> > > How do you install updates?
> > 
> > First I download the new APK, then change it's package name, then install it on top of old app.
> 
> And how do you know there is a new version? You know, you can update with F-Droid.

I have a habit of scrolling new updates section of FDroid, so I'll stumble upon it during scrolling.

**Comment by ololoe** [comment]
> I have a habit of scrolling new updates section of FDroid, so I'll stumble upon it during scrolling.

You have 2 Telegram FOSS, right? If first is updated you can manually update second. What if I have Telegram from google play? How do I know there is an update?

**Comment by ololoe** [comment]
@Gabr-F are there ways to get notified of new version?

**Yes, it is on topic.** Same ids are very uncomfortable.

**Comment by Gabr-F** [comment]
> @Gabr-F are there ways to get notified of new version?
> 
> **Yes, it is on topic.** Same ids are very uncomfortable.

I don't think that @Aryan-Chauhan meant to claim that his method makes same AppIds ok, it was just a temporary workaround he proposed.

**Comment by ololoe** [comment]
Yes, and I meant it's uncomfortable as you have to manually watch for updates.

**Comment by Gabr-F** [comment]
> Yes, and I meant it's uncomfortable as you have to manually watch for updates.

That's most definitely not the biggest inconvenience, and there *are* ways to automatically monitor the updates (I   know of site monitors and F-Droid Build Monitor, but there are probably also tons of apps that can do it)

**Comment by Gabr-F** [comment]
@ololoe And you're the one who sent the f#ck messages right??

That's no attitude towards a person who simply contributes his time for nothing in return, and what the hell did you mean with "links to channels"??

That I'm aware of the only ("only") serious thing that gets lost when you uninstall a Telegram-derived client is the secret chats.

**Comment by ololoe** [comment]
@Gabr-F and make others contribute their time too, whether they want or not, right? :D

Channel properties - create links, or shortcut. My interface language is not english. Creates shortcut on desktop. I had like 50 of them and now they are gone!!!

**Comment by Gabr-F** [comment]
> @Gabr-F and make others contribute their time too, whether they want or not, right? :D

You learned your lesson, that should teach you to stay away from anything you're not paying support for.

You'll probably spare many people a lot of stress if you simply do that.

I hope that those messages were reported to GitHub and you're one step closer to being banned.
 

> Channel properties - create links, or shortcut. My interface language is not english. Creates shortcut on desktop. I had like 50 of them and now they are gone!!!

So freaking home screen shortcuts???

That was your tragedy?

I never heard anyone using that feature, let alone creating 50 shortcuts.

They are handled by Android, I don't think an app can even enumerate them after their creation.

And it's not surprising that they're deleted when you uninstall an application, since they're just shortcuts to the application.

To keep them between uninstalls Telegram would probably have to store somewhere the information that you created them, and create a whole interface for managing them.

It seems dumb since there already are the cloud-synced folders to act as bookmarks; they're admittedly few but that's Telegram's explicit business decision (since you get more with the premium).

If you think that Telegram should warn you that they're not stored between uninstalls, you should tell **them**.

**Comment by ololoe** [comment]
> I hope that those messages were reported to GitHub and you're one step closer to being banned.

Did you confuse github with Facebook? Extinguish your butthurt, noone hears you.

> I never heard anyone using that feature, let alone creating 50 shortcuts.

Who gives a f?

They got deleted because I had to uninstall Telegram because I needed original Telegram to register a new account. That would not happen if FOSS haven't took it's package name.

**Comment by ololoe** [comment]
@Gabr-F I hope your SSD will die some day and you don't have backups. That will be karma. And if you taste salt on your lips it's because I pssed on your face.

**Comment by Gabr-F** [comment]
> @Gabr-F I hope your SSD will die some day and you don't have backups. That will be karma. And if you taste salt on your lips it's because I pssed on your face.

Well I shut up to not get banned as well, and wish you the worst :)

**Comment by mgw2013** [comment]
@Gabr-F Guys, the language used in the recent posts is totally inadequate. There is absolutely no reason to be impolite to anyone here.

**Comment by Aryan-Chauhan** [comment]
> @Gabr-F Guys, the language used in the recent posts is totally inadequate. There is absolutely no reason to be impolite to anyone here.

Are you sure "inadequate" is the correct word to use here?

**Comment by Gabr-F** [comment]
> @Gabr-F Guys, the language used in the recent posts is totally inadequate. There is absolutely no reason to be impolite to anyone here.

Yeah I probably should have curbed myself even more, in theory, but I felt sorry for the developer

**Comment by ololoe** [comment]
@Gabr-F grunt!
